client/dashboard/task-list/tasks/payments/payfast.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import interpolateComponents from 'interpolate-components';
+import { compose } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * WooCommerce dependencies
+ */
+import { Form, Link, TextControl } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { recordEvent } from 'lib/tracks';
+
+class PayFast extends Component {
+	getInitialConfigValues = () => {
+		return {
+			merchant_id: '',
+			merchant_key: '',
+			pass_phrase: '',
+		};
+	};
+
+	validate = ( values ) => {
+		const errors = {};
+
+		if ( ! values.merchant_id ) {
+			errors.merchant_id = __(
+				'Please enter your merchant ID',
+				'woocommerce-admin'
+			);
+		}
+
+		if ( ! values.merchant_key ) {
+			errors.merchant_key = __(
+				'Please enter your merchant key',
+				'woocommerce-admin'
+			);
+		}
+
+		if ( ! values.pass_phrase ) {
+			errors.pass_phrase = __(
+				'Please enter your passphrase',
+				'woocommerce-admin'
+			);
+		}
+
+		return errors;
+	};
+
+	updateSettings = async ( values ) => {
+		const {
+			createNotice,
+			isSettingsError,
+			updateOptions,
+			markConfigured,
+			setRequestPending,
+		} = this.props;
+
+		setRequestPending( true );
+
+		// Because the PayFast extension only works with the South African Rand
+		// currency, force the store to use it
+		await updateOptions( { woocommerce_currency: 'ZAR' } );
```

Comment:
Optional: we can move this into the `updateOptions` call below so we only make a single network request to update options.

Reply:
This is done